### PR TITLE
PRS-3209: improved rendering performance for templates

### PR DESCRIPTION
**Pull request by meyerhp**
## Description
This PR improves the startup and rendering performance of the theme by making the navigation "cachable". I moved the page-specific logic out of the nav templates so Hugo doesn't have to re-render the menu for each page. All of the menu toggle logic has been moved to JS.

See [partialCached](https://gohugo.io/functions/partialcached/) for more info.

## Why?!
Whenever an article is removed, renamed, or added, Hugo needs to recompile all pages because each page has its own menu and all of the menus need to be updated. This takes > 8 seconds for the Span Handbook and will be even slower for  OpenAPI sites with big menus. This PR reduces the startup and re-compile time to <= 1 second.

### Before
```
Template Metrics:

     cumulative       average       maximum         
       duration      duration      duration  count  template
     ----------      --------      --------  -----  --------
  3m41.558139146s    1.388261ms  279.027168ms  159594  partials/nav-item.html
  1m20.667079644s  200.664377ms  472.209203ms    402  partials/navigation.html
  1m8.975290537s  212.231663ms  473.177512ms    325  _default/single.html
  15.308261899s  201.424498ms  413.285029ms     76  _default/list.html
     1.7499081s    1.632376ms   72.618003ms   1072  partials/article.html
   388.052318ms     972.562µs   39.744409ms    399  partials/searchmap-item.html
   340.883995ms     317.988µs   48.694858ms   1072  partials/article-header.html
   169.441137ms  169.441137ms  169.441137ms      1  index.searchmap.json
   150.800753ms     377.946µs   27.352285ms    399  partials/json-nav-item.html
   147.023684ms     152.672µs   48.642174ms    963  partials/author.html
   144.022136ms  144.022136ms  144.022136ms      1  partials/searchmap-root.html
   133.973631ms  133.973631ms  133.973631ms      1  index.html
    93.011738ms    1.310024ms   46.418849ms     71  shortcodes/img.html
    69.970238ms   69.970238ms   69.970238ms      1  index.menuindex.json
    67.901688ms   67.901688ms   67.901688ms      1  partials/json-navigation.html
    59.230881ms      147.34µs   38.406626ms    402  partials/footer.html
    38.950294ms    2.291193ms   11.565524ms     17  shortcodes/tooltip.html
    37.834069ms      39.287µs   25.772858ms    963  partials/status.html
     30.57375ms     122.295µs   27.165528ms    250  shortcodes/baseurl.html
    20.347766ms      21.129µs     3.80264ms    963  partials/roles.html
    17.054684ms      16.028µs   16.334166ms   1064  partials/edit-link.html
    12.115681ms      22.561µs     2.36095ms    537  _default/_markup/render-link.html
     7.317319ms    7.317319ms    7.317319ms      1  _internal/_default/sitemap.xml
     2.265345ms      13.646µs     157.396µs    166  shortcodes/callout.html
     1.102385ms      17.498µs     118.041µs     63  shortcodes/path.html
      523.324µs       1.301µs     197.931µs    402  partials/modal.html
      504.909µs       1.255µs       38.45µs    402  partials/info.html
      121.588µs     121.588µs     121.588µs      1  404.html


                   | EN   
-------------------+------
  Pages            | 404  
  Paginator pages  |   0  
  Non-page files   | 152  
  Static files     |  42  
  Processed images |   0  
  Aliases          |   0  
  Sitemaps         |   1  
  Cleaned          |   0  

Total in 18509 ms
```

### After
```
Template Metrics:

     cumulative       average       maximum         
       duration      duration      duration  count  template
     ----------      --------      --------  -----  --------
   3.034267685s    9.336208ms  292.694891ms    325  _default/single.html
   1.756149152s     552.943µs   75.478113ms   3176  partials/nav-item.html
   1.277512059s    3.123501ms  128.189744ms    409  partials/navigation.html
   893.504981ms     833.493µs   52.027219ms   1072  partials/article.html
   384.933085ms    5.064909ms   91.839996ms     76  _default/list.html
   237.261155ms     594.639µs   19.955123ms    399  partials/searchmap-item.html
   136.566684ms     127.394µs   12.915329ms   1072  partials/article-header.html
    97.026369ms   97.026369ms   97.026369ms      1  _internal/_default/sitemap.xml
    92.022268ms   92.022268ms   92.022268ms      1  index.searchmap.json
    85.822382ms   85.822382ms   85.822382ms      1  partials/searchmap-root.html
    35.666947ms      37.037µs   12.840104ms    963  partials/roles.html
    32.429695ms    1.907629ms    9.780907ms     17  shortcodes/tooltip.html
    23.157879ms      43.124µs    4.208245ms    537  _default/_markup/render-link.html
    22.849735ms      23.727µs    6.463593ms    963  partials/status.html
    21.383968ms      22.205µs     1.45026ms    963  partials/author.html
    13.648422ms     192.231µs    7.263332ms     71  shortcodes/img.html
     5.304063ms       4.985µs    2.031584ms   1064  partials/edit-link.html
     3.572518ms       14.29µs      95.053µs    250  shortcodes/baseurl.html
     2.341922ms      14.107µs      92.355µs    166  shortcodes/callout.html
     1.522822ms      24.171µs     112.976µs     63  shortcodes/path.html
      758.351µs       1.886µs      603.68µs    402  partials/modal.html
      468.183µs       1.164µs     225.497µs    402  partials/footer.html
       316.96µs      316.96µs      316.96µs      1  index.html
      263.193µs         654ns      26.965µs    402  partials/info.html
       71.395µs      71.395µs      71.395µs      1  404.html

Total in 1210 ms
````

### Issue
https://spandigital.atlassian.net/browse/PRSDM-3209

### Testing
The menu should work exactly the same as before  but running `hugo` should be a lot quicker.

### Checklist before merging

* [x] Did you test your changes locally?
* [x] Did you update the CHANGELOG?
* [x] Is the documentation updated for this change? (If Required)

### CODE REVIEW

File: layouts/partials/nav-item.html [hunk 1]
```diff
@@ -1,29 +1,9 @@
 {{ $childLevel := add .Level 1 }}
-{{ $currentPage := .CurrentPage }}
+{{/*  {{ $currentPage := .CurrentPage }}  */}}
```
Comment (christopherbrunsdon) [inline layouts/partials/nav-item.html:2]: Is this intentional to comment out, or are you still going to delete?

**Comment by christopherbrunsdon** [comment]
While testing, I'm getting unexpected behaviour on the side menu for handbook. See video.

When clicking links, it closes the nested lists unexpectedly. When I also click to expand a nested-nested list, it closes.

Clicking on the text vs arrow gives me different behaviour.

content-builder is on branch `develop` and I `rm` the tmp/cache.
 

https://user-images.githubusercontent.com/1186788/211751947-1567a141-adc7-446f-9f93-24ac8b04bf24.mov

File: layouts/partials/nav-item.html (hunk 1 above)
Comment (meyerhp) [reply layouts/partials/nav-item.html:2]: No, I forgot to remove this. Will update. Thanks!

**Review by christopherbrunsdon — APPROVED** [review]
Tested ... OK

Approved JS PR.

File: static/presidium.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
...tialState(): must return an object or null",t.displayName||"ReactCompositeComponent"),this.state=a};for(var r in t.prototype=new x,t.prototype.constructor=t,t.prototype.__reactAutoBindPairs=[],a.forEach(p.bind(null,t)),p(t,v),p(t,e),p(t,g),t.getDefaultProps&&(t.defaultProps=t.getDefaultProps()),i(t.prototype.render,"createClass(...): Class specification must implement a `render` method."),u)t.prototype[r]||(t.prototype[r]=null);return t}}},139:e=>{"use strict";function t(e){return function(){return e}}var n=function(){};n.thatReturns=t,n.thatReturnsFalse=t(!1),n.thatReturnsTrue=t(!0),n.thatReturnsNull=t(null),n.thatReturnsThis=function(){return this},n.thatReturnsArgument=function(e){return e},e.exports=n},3677:e=>{"use strict";e.exports={}},3759:e=>{"use strict";e.exports=function(e,t,n,r,o,i,s,a){if(!e){var u;if(void 0===t)u=new Error("Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings.");else{var c=[n,r,o,i,s,a],l=0;(u=new Error(t.replace(/%s/g,(function(){return c[l++]})))).name="Invariant Violation"}throw u.framesToPop=1,u}}},3620:(e,t,n)=>{"use strict";var r=n(139);e.exports=r},9755:function(e,t){var n;
```
Comment (effleurager) [inline static/presidium.js:1]: Isn't this file covered by https://github.com/SPANDigital/presidium-js/pull/100?

Comment (meyerhp) [reply static/presidium.js:1]: Yes, but it needs to be updated in the theme as well.

File: layouts/partials/nav-item.html (hunk 1 above)
Comment (meyerhp) [reply layouts/partials/nav-item.html]: Fixed

**Review by effleurager — APPROVED** [review]

**Pull request merged**